### ISSUE TITLE
Enable gateways to subscribe for commands of a specific device

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#1523]
With the implementation of #930, we have added support for gateways creating command subscriptions without having to use individual device ids. 

The gateway will for example just create a MQTT subscription or AMQP receiver link on the `command` topic/link address. Northbound applications will send commands with the specific device id in the message address and the gateway mapping logic (based on the `lastKnownGateway` information for the device) will forward such commands to the corresponding gateway.

*What is not currently possible* in such a scenario is that the applications will get notified of when a particular device is ready to receive commands.

That however is a cornerstone of the Hono command & control feature in how it is defined for non-gateway devices.

In order to support this principle also for gateways, the command handling logic in the protocol adapters should be adapted to enable gateways to alternatively subscribe for commands of a specific device.
Hono currently doesn't support this because there would be *device*-specific AMQP consumers created in the protocol adapter, whereas the gateway mapping logic would always redirect the command to a *gateway*-specific consumer.

When receiving such a command subscription with a device id from a gateway, the protocol adapter should forward a corresponding notification with the device id to the northbound application.

To sum it up, the goal here would be to have gateways be able to
- either create a single subscription for commands regarding *all devices* that are currently mapped to the gateway (there will be no individual device-ready-for-commands notification here)
- or create individual command subscriptions for specific devices - with corresponding downstream notifications.

---
As to the implementation of this:
Even though the device id is given with the subscription request, the protocol adapter shall create/use a gateway-specific command consumer and map the command received on that consumer to the original device of that command.

**Comment by sophokles73** [comment]
@calohmn my understanding is that this is the overall issue for dding this functionality to *all* adapters, i.e. there will be subsequent PRs for adding the functionality to HTTP, MQTT, AMQP, right?

**Comment by calohmn** [comment]
@sophokles73 Right, I'm about to create a PR for that.

**Comment by calohmn** [comment]
@sophokles73 I've added the relevant PRs for
- AMQP/MQTT/HTTP adapter changes
- Integration tests
- HTTP adapter change to include the target device id in the HTTP headers of the response with the command
- Documentation / release notes

**Comment by calohmn** [comment]
All corresponding PRs have been merged now.
